packages/connectivity/lib/connectivity.dart

```diff
@@ -95,14 +175,27 @@ class Connectivity {
   }
 }
 
-ConnectivityResult _parseConnectivityResult(String state) {
-  switch (state) {
+ConnectionSubtype _parseConnectionSubtype(String state) {
+  return connectionTypeMap[state];
+}
+
+ConnectivityInfo _parseConnectivityResult(String state) {
```

Comment:
let's keep this method the same and add another method which calls this and `_parseConnectionSubtype` to parse the ConnectivityDetailedResult
maybe call it
```dart
ConnectivityDetailedResult _parseConnectivityDetailedResult(String state);
```